src/nanodbc.cpp

```diff
@@ -994,15 +994,15 @@ class connection::connection_impl
         return string_type(&name[0], &name[strarrlen(name)]);
     }
 
-
     std::size_t ref_transaction()
     {
-        return --transactions_;
+        return ++transactions_;
     }
 
     std::size_t unref_transaction()
     {
-        return ++transactions_;
+        NANODBC_ASSERT(transactions_ != 0);
```

Comment:
If I changed the code to `if (transactions_ > 0) --transactions_;` it would be
* change in the original behaviour.
* unrelated to this particular bug fix

both I prefered to avoid and I used assertion to *comment* some obvious expectations.

I will replace it with the `if-condition`.